scripts/binary_to_array.py

```diff
@@ -218,11 +218,30 @@ def source(namespaces, array_name, array_size_name, fileid, filename,
   return data
 
 
+def _get_repo_root(filename):
+  """Returns the root of the repository containing the given file.
+
+  Args:
+    filename: A source file or directory in the repository.
+  """
+
+  # CMake builds can be run out of source tree, so use the directory containing
+  # a source file as the starting point for the calculation.
+  if not os.path.isdir(filename):
```

Comment:
Nit: I think it would be slightly more readable if this used a new intermediate variable (I would expect `filename` to be a "leaf" file, not a directory). Consider something like:
```python
dir_in_repo = filename if os.path.isdir(filename) else os.path.dirname(filename)
```

Reply:
The one-liner doesn't fit in the allowed line length, so I split it across multiple lines